internal/cli/action/customize.go

```diff
@@ -18,97 +18,146 @@ limitations under the License.
 package action
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"os/signal"
+	"path/filepath"
 	"syscall"
+	"time"
 
 	"github.com/urfave/cli/v2"
 
 	"github.com/suse/elemental/v3/internal/cli/cmd"
-	"github.com/suse/elemental/v3/pkg/deployment"
-	"github.com/suse/elemental/v3/pkg/installer"
+	"github.com/suse/elemental/v3/internal/config"
+	"github.com/suse/elemental/v3/internal/customize"
+	"github.com/suse/elemental/v3/internal/image"
+	"github.com/suse/elemental/v3/pkg/extractor"
+	"github.com/suse/elemental/v3/pkg/helm"
+	"github.com/suse/elemental/v3/pkg/http"
 	"github.com/suse/elemental/v3/pkg/sys"
+	"github.com/suse/elemental/v3/pkg/sys/platform"
+	"github.com/suse/elemental/v3/pkg/sys/vfs"
 )
 
-func Customize(ctx *cli.Context) error {
+func Customize(ctx *cli.Context) (err error) {
 	if ctx.App.Metadata == nil || ctx.App.Metadata["system"] == nil {
 		return fmt.Errorf("error setting up initial configuration")
 	}
-	s := ctx.App.Metadata["system"].(*sys.System)
+	system := ctx.App.Metadata["system"].(*sys.System)
+	logger := system.Logger()
 	args := &cmd.CustomizeArgs
-	logger := s.Logger()
 
 	ctxCancel, cancelFunc := signal.NotifyContext(ctx.Context, syscall.SIGTERM, syscall.SIGINT)
 	defer cancelFunc()
 
 	logger.Info("Customizing image")
 
-	mType, err := installer.StringToMediaType(args.Type)
+	customizeDir := fmt.Sprintf("customize-%s", time.Now().UTC().Format("2006-01-02T15-04-05"))
+	outDir, err := config.CreateOutputDir(system.FS(), args.CustomizeOutput, customizeDir, 0700)
 	if err != nil {
-		return fmt.Errorf("invalid media type for customize: %w", err)
+		logger.Error("Creating customize directory '%s' failed", outDir)
+		return err
 	}
-	media := installer.NewMedia(ctxCancel, s, mType)
 
-	digestCustomizeSetup(args, media)
+	defer func() {
+		logger.Debug("Cleaning up customize-dir %s", outDir)
+		rmErr := system.FS().RemoveAll(string(outDir))
+		if rmErr != nil {
+			logger.Error("Cleaning up customize-dir %s", outDir)
+			err = errors.Join(err, rmErr)
+		}
+	}()
 
-	d, err := digestCustomizeDeploymentSetup(s, args)
+	def, err := digestCustomizeDefinition(system.FS(), args)
 	if err != nil {
-		s.Logger().Error("Failed to collect deployment setup")
+		logger.Error("Digesting image definition from customize flags")
```

Comment:
```suggestion
		logger.Error("Digesting image definition from customize flags failed")
```

Many of the other `.Error()` logs also don't contain "failed" or "error" and it reads like an info log but on the error level. Could we fix these too?